packages/ui/src/utils/ssr-utils.ts

```diff
@@ -1,2 +1,4 @@
 export const isServer = () => typeof window === 'undefined'
 export const isClient = () => !isServer()
+export const getWindow = (): Window | undefined => typeof window === 'undefined' ? undefined : window
+export const getDocument = (): Document | undefined => typeof document === 'undefined' ? undefined : document
```

Comment:
What does it do? Isn't it the same as calling window and document directly and either getting them or undefined?

Reply:
In SSR you can not access window. It is part of browser's DOM. So you need first check if it defined, then access. We care about it in codes that run before components mounted.